CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();
+    DoSomerthingLater();
+}
+```
+
+### Remove unused or redudant namespaces
+
+### Use single line and lambda getters
+
+```csharp
+// Good
+public string Name { get; private set; }
+
+// Good
+public string UppercaseName => Name.toUpperCase(); 
+```
+
+### Don't use braces for just one line
+
+```csharp
+// Bad
+if (payment == null)
+{
+    return "A payment is required.";
+}
+
+// Good
+if (payment == null)
+    return "A payment is required.";
+```
+
+### Use spaces in array/object initializers
+
+```csharp
+// Bad
+var names = new string[] {"Marie","John","Paul"};
+
+// Bad
+var names = new string[] {"Marie", "John", "Paul"};
+
+// Good
+var names = new string[] { "Marie", "John", "Paul" };
+```
+
+### Use spaces between operators
+
+```csharp
+// Bad
+var result = 4/2;
+
+// Bad
+var result = 4/2+6;
+
+// Good
+var result = 4 / 2;
+
+// Good
+var result = (4 / 2) + 6;

```

Comment:
To avoid writing things like ' Code must not contain space after new keyword in implicitly typed array allocation'

I think we should add a comprehensive piece of code that have examples about every spacing scenario and add a single rule saying:

_Spacing should comply with StyleCop rules SA1001 to SA1026_